lib/lrama/state.rb

```diff
@@ -142,5 +160,274 @@ def rr_conflicts
         conflict.type == :reduce_reduce
       end
     end
+
+    def propagate_lookaheads(next_state)
+      next_state.kernels.map {|item|
+        lookahead_sets =
+          if item.position == 1
```

Comment:
[note]: Corresponding to case 2 of Definition 3.40 because `position` is 0-origin in this code base.